internal/pkg/deploy/cloudformation/stack/backend_svc_test.go

```diff
@@ -217,41 +217,45 @@ Outputs:
 				m.EXPECT().Read(albRulePriorityGeneratorPath).Return(&template.Content{Buffer: bytes.NewBufferString("something")}, nil)
 				m.EXPECT().Read(desiredCountGeneratorPath).Return(&template.Content{Buffer: bytes.NewBufferString("something")}, nil)
 				m.EXPECT().Read(envControllerPath).Return(&template.Content{Buffer: bytes.NewBufferString("something")}, nil)
-				m.EXPECT().ParseBackendService(template.WorkloadOpts{
-					WorkloadType: manifest.BackendServiceType,
-					HealthCheck: &template.ContainerHealthCheck{
-						Command:     []string{"CMD-SHELL", "curl -f http://localhost/ || exit 1"},
-						Interval:    aws.Int64(5),
-						Retries:     aws.Int64(3),
-						StartPeriod: aws.Int64(0),
-						Timeout:     aws.Int64(10),
-					},
-					HostedZoneAliases: make(template.AliasesForHostedZone),
-					HTTPHealthCheck: template.HTTPHealthCheckOpts{
-						HealthCheckPath: manifest.DefaultHealthCheckPath,
-						GracePeriod:     aws.Int64(manifest.DefaultHealthCheckGracePeriod),
-					},
-					DeregistrationDelay: aws.Int64(60), // defaults to 60
-					RulePriorityLambda:  "something",
-					DesiredCountLambda:  "something",
-					EnvControllerLambda: "something",
-					ExecuteCommand:      &template.ExecuteCommandOpts{},
-					NestedStack: &template.WorkloadNestedStackOpts{
-						StackName:       addon.StackName,
-						VariableOutputs: []string{"MyTable"},
-					},
-					Network: template.NetworkOpts{
-						AssignPublicIP: template.DisablePublicIP,
-						SubnetsType:    template.PrivateSubnetsPlacement,
-						SecurityGroups: []string{"sg-1234"},
-					},
-					DeploymentConfiguration: template.DeploymentConfigurationOpts{
-						MinHealthyPercent: 0,
-						MaxPercent:        100,
-					},
-					EntryPoint: []string{"enter", "from"},
-					Command:    []string{"here"},
-				}).Return(&template.Content{Buffer: bytes.NewBufferString("template")}, nil)
+				m.EXPECT().ParseBackendService(gomock.Any()).DoAndReturn(func(actual template.WorkloadOpts) (*template.Content, error) {
+					require.Equal(t, template.WorkloadOpts{
+						WorkloadType: manifest.BackendServiceType,
+						HealthCheck: &template.ContainerHealthCheck{
+							Command:     []string{"CMD-SHELL", "curl -f http://localhost/ || exit 1"},
+							Interval:    aws.Int64(5),
+							Retries:     aws.Int64(3),
+							StartPeriod: aws.Int64(0),
+							Timeout:     aws.Int64(10),
+						},
+						HostedZoneAliases: make(template.AliasesForHostedZone),
+						HTTPHealthCheck: template.HTTPHealthCheckOpts{
+							HealthCheckPath: manifest.DefaultHealthCheckPath,
+							GracePeriod:     aws.Int64(manifest.DefaultHealthCheckGracePeriod),
+						},
+						DeregistrationDelay: aws.Int64(60), // defaults to 60
+						CustomResources:     make(map[string]template.S3ObjectLocation),
```

Comment:
The only thing added is this field. I've just changed the tests to use `DoAndReturn` and use `require.Equal` to compare objects as it works better than `gomock`.

`Gomock` would return an error when the field's location is different than expected

Reply:
great change this was so tedious before lol